PiwikTracker/Device.swift

```diff
@@ -0,0 +1,132 @@
+            
+        // iPod
+        case "iPod1,1":      return "iPod Touch 1G"
+        case "iPod2,1":      return "iPod Touch 2G"
+        case "iPod3,1":      return "iPod Touch 3G"
+        case "iPod4,1":      return "iPod Touch 4G"
+        case "iPod5,1":      return "iPod Touch 5G"
+        case "iPod6,1":      return "iPod Touch 6G"
+            
+        // iPad
+        case "iPad1,1":      return "iPad 1"
+        case "iPad2,1":      return "iPad 2 (WiFi)"
+        case "iPad2,2":      return "iPad 2 (GSM)"
+        case "iPad2,3":      return "iPad 2 (CDMA)"
+        case "iPad2,4":      return "iPad 2 (WiFi)"
+        case "iPad2,5":      return "iPad Mini 1 (WiFi)"
+        case "iPad2,6":      return "iPad Mini 1 (GSM)"
+        case "iPad2,7":      return "iPad Mini 1 (GSM+CDMA)"
+        case "iPad3,1":      return "iPad 3 (WiFi)"
+        case "iPad3,2":      return "iPad 3 (GSM+CDMA)"
+        case "iPad3,3":      return "iPad 3 (GSM)"
+        case "iPad3,4":      return "iPad 4 (WiFi)"
+        case "iPad3,5":      return "iPad 4 (GSM)"
+        case "iPad3,6":      return "iPad 4 (GSM+CDMA)"
+        case "iPad4,1":      return "iPad Air 1 (WiFi)"
+        case "iPad4,2":      return "iPad Air 1 (Cellular)"
+        case "iPad4,3":      return "iPad Air"
+        case "iPad4,4":      return "iPad Mini 2 (WiFi)"
+        case "iPad4,5":      return "iPad Mini 2 (Cellular)"
+        case "iPad4,6":      return "iPad Mini 2 (Rev)"
+        case "iPad4,7":      return "iPad Mini 3 (WiFi)"
+        case "iPad4,8":      return "iPad Mini 3 (A1600)"
+        case "iPad4,9":      return "iPad Mini 3 (A1601)"
+        case "iPad5,1":      return "iPad Mini 4 (WiFi)"
+        case "iPad5,2":      return "iPad Mini 4 (Cellular)"
+        case "iPad5,3":      return "iPad Air 2 (WiFi)"
+        case "iPad5,4":      return "iPad Air 2 (Cellular)"
+        case "iPad6,3":      return "iPad Pro 9.7 (WiFi)"
+        case "iPad6,4":      return "iPad Pro 9.7 (Cellular)"
+        case "iPad6,7":      return "iPad Pro 12.9 (WiFi)"
+        case "iPad6,8":      return "iPad Pro 12.9 (Cellular)"
+            
+        case "i386":         return "Simulator"
+        case "x86_64":       return "Simulator"
+            
+        default: return platform
+        }
+    }
+    
+    /// Reaturns the version number of the current OS as String i.e. "1.2" or "9.4"
+    private static func currentOSVersion() -> String  {
+        return UIDevice.current.systemVersion
+    }
+    
+    // Returns the screen size in points
+    private static func currentScreenSize() ->  String {
+        let bounds = UIScreen.main.bounds
+        return "\(bounds.size.width)x\(bounds.size.height)"
+    }
+    
+    // Returns the screen size in pixels
+    private static func currentNativeScreenSize() ->  String {
+        let bounds = UIScreen.main.nativeBounds
+        return "\(bounds.size.width)x\(bounds.size.height)"
+    }
```

Comment:
Those static methods with a `current` prefixed to irritate me a bit. A possible alternative would be a static method on `Device` that returns a new Device populated with values for the current device. What do you think?